tests/app/submitter/test_submitter.py

```diff
@@ -77,7 +77,22 @@ def test_when_fail_to_publish_message_then_returns_false(self):
         connection.channel.side_effect = Mock(return_value=channel)
         with patch('app.submitter.submitter.BlockingConnection', return_value=connection):
             # When
-            published = self.submitter.send_message(message={}, queue=self.queue_name)
+            published = self.submitter.send_message(message={}, queue=self.queue_name, tx_id='12345')
 
             # Then
             self.assertFalse(published, 'send_message should fail to publish message')
+
+    def test_when_message_sent_then_tx_id_is_sent_in_header(self):
```

Comment:
It shouldn't be None in our normal flow as tx_id has a generator. But i have added a test and the functionality so that we can call `send_message` without one.